idea/src/org/jetbrains/kotlin/idea/quickfix/TooLongCharLiteralToStringFix.kt

```diff
@@ -0,0 +1,40 @@
+package org.jetbrains.kotlin.idea.quickfix
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.diagnostics.Errors
+import org.jetbrains.kotlin.psi.KtConstantExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtPsiFactory
+
+class TooLongCharLiteralToStringFix(
+        element: KtConstantExpression
+) : KotlinQuickFixAction<KtConstantExpression>(element) {
+    override fun getText(): String = "Convert too long character literal to string"
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val text = element.text
+        if (!(text.startsWith("'") && text.endsWith("'") && text.length >= 2)) {
+            return
+        }
+
+        val newStringContent = text
+                .slice(1..text.length - 2)
+                .replace("\\", "\\\\")
+                .replace("\"", "\\\"")
```

Comment:
Changing double quote to escaped double quote is logical, I think it should be left as is.